packages/builtin/array.pony

```diff
@@ -433,15 +433,17 @@ class Array[A] is Seq[A]
       end
     end
 
-  fun iso chop(split_point: USize): (Array[A] iso^, Array[A] iso^) =>
+  fun iso chop[B: (A & Any #send) = A](split_point: USize): (Array[A] iso^, Array[A] iso^) =>
     """
     Chops the array in half at the split point requested and returns both
     the left and right portions. The original array is trimmed in place and
     returned as the left portion. If the split point is larger than the
     array, the left portion is the original array and the right portion
     is a new empty array.
-    Both arrays are isolated and mutable, as they do not share memory.
     The operation does not allocate a new array pointer nor copy elements.
+
+    The entry type must be sendable so that the two array pieces can
+    be deeply isolated from each other.
```

Comment:
I think this could use some elaboration; the example you gave in the issue would be good here as not everyone is going to understand what this means.